packages/bot-skeleton/src/services/tradeEngine/trade/Balance.js

```diff
@@ -14,8 +14,7 @@ export default Engine =>
                     } = data;
 
                     balance_string = getFormattedText(b, currency);
-
-                    info({ accountID: this.accountInfo.loginid, balance: balance_string });
+                    info({ accountID: this.accountInfo?.loginid, balance: balance_string });
```

Comment:
In DBot there is a problem.
With every request such as if we request for tick, then in the response there will be a tick object or error in worst case.
So here we are checking for the `msg_type === 'authorize'` and setting the whole data (response) object in the accountInfo which I think is incorrect.

bot-skeleton>src>services>tradeEngine>trade>index.js loginAndGetBalance()

![Screenshot 2022-09-23 at 12 10 12 PM](https://user-images.githubusercontent.com/90243468/191891086-9d5f5ffe-d791-40df-a52f-6fae94c704a2.png)

This is the response object of authorize.
the whole object is data.

<img width="922" alt="Screenshot 2022-09-23 at 12 18 21 PM" src="https://user-images.githubusercontent.com/90243468/191891337-fe4aa29a-cd86-4619-8c4d-b9d338f3da66.png">

Now the places where we are using `this.accountInfo.loginId` it will always give error or undefined because the data object doesn't contain `loginid` but `authorize` inside data does.

and here you can see the result of console which I added inside trade>balance.js

![Screenshot 2022-09-23 at 12 24 39 PM](https://user-images.githubusercontent.com/90243468/191891963-838d2f5b-e486-403f-b467-792e68a65fd5.png)

and this all is happening in `master` branch.

So I just corrected the assignment of object in trade>index.js file. And it works fine now
@sara-fs